interp/value.go

```diff
@@ -214,6 +214,31 @@ func genValue(n *node) func(*frame) reflect.Value {
 	}
 }
 
+func genDestValue(typ *itype, n *node) func(*frame) reflect.Value {
+	convertLiteralValue(n, typ.TypeOf())
```

Comment:
I see that convertLiteralValue was called in arrayLit, mapLit, and send, so I understand why it is now in genDestValue. But it was not called in assign, and it will now be called in assign since genDestValue now replaces the big switch of assign. Is it ok to call it even though it was not called before?

Reply:
I haven't seen regressions due to that, and at this stage, I favour code factoring, having most of the tests at one place, rather than many places where several variants of mostly similar tests are done, which has proved a source of bugs.

Reply:
fair enough